### ISSUE TITLE
pocl 0.14 install fails

### ISSUE THREAD

**Issue by ax3l** [spack/spack#4669]
Building OpenCL via `pocl` fails on creating a symlink as of e5560dfbdee0235f6f727071cec968fd4913fab4

```bash
spack install pocl

==> Building pocl [CMakePackage]
==> Executing phase : 'cmake'
==> Executing phase : 'build'
==> Executing phase : 'install'
==> Error: OSError: [Errno 17] File exists
/home/axel/src/spack/var/spack/repos/builtin/packages/pocl/package.py:95, in symlink_opencl:
     92       @run_after('install')
     93       def symlink_opencl(self):
     94           with working_dir(self.build_directory):
  >> 95               os.symlink("OpenCL", join_path(self.prefix.include, "CL"))
```

The stage looks like this:
```bash
$ ls include/CL/
cl2.hpp  cl_egl.h  cl_ext.h  cl_gl_ext.h  cl_gl.h  cl.h  cl.hpp  cl.hpp.in  cl.hpp.patched  cl_platform.h  CMakeLists.txt  opencl.h
$ ls include/OpenCL/
cl_ext.h  cl_gl_ext.h  cl_gl.h  cl.h  cl.hpp  cl_platform.h  opencl.h
```

But in the installed directory the `OpenCL` symlink is missing if I blindly comment out those lines.

**Comment by ax3l** [comment]
@eschnett as the original author, do you have any idea why this fails?

**Comment by eschnett** [comment]
Did you install from scratch, or was this attempt to continue a previously aborted install? The latter could explain why the symlink already exists.

**Comment by ax3l** [comment]
Thanks for the hint.
The issue also appears when building from scratch (on a Debian testing with Python 2.7.13 and nvidia drivers installed).

**Comment by eschnett** [comment]
I tried to reproduce this on an Ubuntu 14.04 system, but got stuck at installing LLVM, which is a prerequisite for pocl. The error is something like
```
unrecognized relocation (0x2a)
```

**Comment by eschnett** [comment]
The symlink is relative. The current working directory is irrelevant. My original install code was refactored at some point, and apparently a now unnecessary `with working_dir(self.build_directory)` was left around. The symlink is supposed to be in the install tree, in the `include` subdirectory, and point from `CL` to `OpenCL`.

Can you check whether such a symlink already exists in your install tree? You have to build with `--keep-prefix` to check this. If so, the work-around is to not flag this as failure. If not, we need to find another way of creating this symlink.

Apologies for debugging this indirectly.

**Comment by ax3l** [comment]
> I tried to reproduce this [...]

[Here](https://github.com/ax3l/dockerfiles/tree/master/spack/opencl) is a Ubuntu 16.04 Dockerfile to reproduce the issue:

**Dockerfile:**
```docker
FROM       ubuntu:16.04

# general environment for docker
ENV        HOME /home
ENV        DEBIAN_FRONTEND noninteractive

# spack environment
ENV        PATH "$HOME/src/spack/bin:$PATH"
COPY       packages.yaml $HOME/.spack/packages.yaml

# install minimal spack dependencies
RUN        apt-get update \
           && apt-get install -y --no-install-recommends \
              autoconf \
              ca-certificates \
              cmake \
              curl \
              git \
              build-essential \
              pkg-config \
              python \
              nano \
              unzip \
           && rm -rf /var/lib/apt/lists/*

# install spack
RUN        git clone --depth 50 https://github.com/llnl/spack.git \
           $HOME/src/spack/

# install PIConGPU dependencies
RUN        spack install pocl
```

**packages.yaml** (place in same dir):
```yaml
packages:
  pkg-config:
    paths:
      pkg-config@0.29.2%gcc@5.4.0 arch=linux-ubuntu16-x86_64: /usr
    buildable: False
  cmake:
    paths:
      cmake@3.5.1%gcc@5.4.0 arch=linux-ubuntu16-x86_64: /usr
    buildable: False
```

Reproduce:
```bash
docker build -t spack_pocl .
```

**Comment by ax3l** [comment]
> Can you check whether such a symlink already exists in your install tree? You have to build with --keep-prefix to check this

will do!

**Comment by alalazo** [comment]
Claimed to be fixed by #6372. Can you confirm @ax3l ?

**Comment by ax3l** [comment]
Wonderful, it works! :)
